lib/host/events/submit_all_orders.js

```diff
@@ -58,7 +59,19 @@ module.exports = async (aoHost, gid, orders) => {
     if (!order.meta) order.meta = {}
 
     if (_isFunction(genOrderLabel)) {
-      order.meta.label = genOrderLabel(state)
+      const label = genOrderLabel(state)
+
+      if (_isString(label)) {
+        order.meta.label = label
+      } else if (_isObject(label) && _isString(label.origin)) {
```

Comment:
@vigan-abd only in this case or on line 21 too?

```js
  const _orders = _isObject(orders)
    ? Object.values(orders)
    : orders
```
and after it is used as array 
```js
 _orders.reduce((obj, ///
_orders.map(///
```

Probably the ternary operator is useless here, because `_isObject(orders)` always `true`, regardless of whether the parameter is an object or an array

Reply:
fixed